In [1]:
import datasets
import obscure_questions
from tqdm.auto import tqdm

c:\Users\gardy\CS4973\travel_agent\.venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
nuwiki = datasets.load_dataset("nuprl/engineering-llm-systems", "wikipedia-northeastern-university", split="test")
questions_tiny = datasets.load_dataset("nuprl/engineering-llm-systems", "obscure_questions", split="tiny")

In [3]:
rag = obscure_questions.RAG(nuwiki)

In [7]:
def get_problem_accuracy(problems) -> float:
    num_correct = 0
    for i, problem in enumerate(tqdm(problems)):
        docs = rag.combined_rank(problem['prompt'])
        result = obscure_questions.answer_query(problem['prompt'], problem['choices'], docs)
        if result == problem['correct_answer']:
            num_correct += 1
        else:
            print(f'Expected: {problem["correct_answer"]}')
            print(f'Got: {result}')
        print(f'{num_correct} / {i + 1}')

    return num_correct / len(problems)

get_problem_accuracy(questions_tiny)

  2%|▏         | 1/50 [00:03<03:05,  3.79s/it]

1 / 1


  4%|▍         | 2/50 [00:07<03:11,  4.00s/it]

Expected: B
Got: A
1 / 2


  6%|▌         | 3/50 [00:11<02:59,  3.83s/it]

2 / 3


  8%|▊         | 4/50 [00:16<03:10,  4.14s/it]

Expected: B
Got: A
2 / 4


 10%|█         | 5/50 [00:21<03:31,  4.69s/it]

Expected: B
Got: C
2 / 5


 12%|█▏        | 6/50 [00:27<03:39,  4.99s/it]

3 / 6


c:\Users\gardy\CS4973\obscure_questions\obscure_questions.py:84: RuntimeWarning: invalid value encountered in divide
  normalized_vec = vec / np.linalg.norm(vec)
 14%|█▍        | 7/50 [00:33<03:43,  5.19s/it]

Expected: B
Got: C
3 / 7


 16%|█▌        | 8/50 [00:38<03:45,  5.37s/it]

4 / 8


 18%|█▊        | 9/50 [00:44<03:45,  5.49s/it]

5 / 9


 20%|██        | 10/50 [00:50<03:41,  5.53s/it]

Expected: C
Got: D
5 / 10


 22%|██▏       | 11/50 [00:55<03:29,  5.36s/it]

6 / 11


 24%|██▍       | 12/50 [01:00<03:23,  5.35s/it]

7 / 12


 26%|██▌       | 13/50 [01:05<03:18,  5.36s/it]

Expected: B
Got: A
7 / 13


 28%|██▊       | 14/50 [01:11<03:15,  5.44s/it]

8 / 14


 30%|███       | 15/50 [01:16<03:05,  5.29s/it]

Expected: C
Got: B
8 / 15


 32%|███▏      | 16/50 [01:21<02:59,  5.28s/it]

9 / 16


 34%|███▍      | 17/50 [01:27<02:56,  5.35s/it]

Expected: B
Got: D
9 / 17


 36%|███▌      | 18/50 [01:32<02:53,  5.44s/it]

Expected: C
Got: B
9 / 18


 38%|███▊      | 19/50 [01:38<02:46,  5.38s/it]

Expected: A
Got: D
9 / 19


 40%|████      | 20/50 [01:43<02:38,  5.29s/it]

Expected: C
Got: A
9 / 20


 42%|████▏     | 21/50 [01:48<02:31,  5.22s/it]

Expected: A
Got: B
9 / 21


 44%|████▍     | 22/50 [01:52<02:21,  5.07s/it]

10 / 22


 46%|████▌     | 23/50 [01:58<02:22,  5.27s/it]

11 / 23


 48%|████▊     | 24/50 [02:04<02:18,  5.32s/it]

Expected: C
Got: B
11 / 24


 50%|█████     | 25/50 [02:09<02:13,  5.35s/it]

12 / 25


 52%|█████▏    | 26/50 [02:15<02:09,  5.40s/it]

13 / 26


 54%|█████▍    | 27/50 [02:19<01:57,  5.11s/it]

14 / 27


 56%|█████▌    | 28/50 [02:24<01:52,  5.09s/it]

15 / 28


 58%|█████▊    | 29/50 [02:30<01:50,  5.28s/it]

16 / 29


 60%|██████    | 30/50 [02:36<01:48,  5.43s/it]

17 / 30


 62%|██████▏   | 31/50 [02:42<01:47,  5.65s/it]

18 / 31


 64%|██████▍   | 32/50 [02:48<01:43,  5.77s/it]

Expected: B
Got: A
18 / 32


 66%|██████▌   | 33/50 [02:54<01:38,  5.77s/it]

Expected: B
Got: A
18 / 33


 68%|██████▊   | 34/50 [02:59<01:31,  5.72s/it]

Expected: C
Got: B
18 / 34


 70%|███████   | 35/50 [03:04<01:23,  5.57s/it]

19 / 35


 72%|███████▏  | 36/50 [03:10<01:19,  5.67s/it]

Expected: C
Got: B
19 / 36


 74%|███████▍  | 37/50 [03:14<01:07,  5.23s/it]

Expected: B
Got: A
19 / 37


 76%|███████▌  | 38/50 [03:20<01:04,  5.40s/it]

Expected: B
Got: C
19 / 38


 78%|███████▊  | 39/50 [03:25<00:58,  5.35s/it]

20 / 39


 80%|████████  | 40/50 [03:31<00:54,  5.43s/it]

21 / 40


 82%|████████▏ | 41/50 [03:37<00:49,  5.46s/it]

22 / 41


 84%|████████▍ | 42/50 [03:42<00:44,  5.51s/it]

23 / 42


 86%|████████▌ | 43/50 [03:48<00:39,  5.59s/it]

24 / 43


 88%|████████▊ | 44/50 [03:54<00:33,  5.63s/it]

25 / 44


 90%|█████████ | 45/50 [03:59<00:28,  5.64s/it]

Expected: C
Got: D
25 / 45


 92%|█████████▏| 46/50 [04:05<00:22,  5.52s/it]

Expected: B
Got: A
25 / 46


 94%|█████████▍| 47/50 [04:10<00:16,  5.48s/it]

Expected: B
Got: A
25 / 47


 96%|█████████▌| 48/50 [04:16<00:11,  5.62s/it]

Expected: B
Got: C
25 / 48


 98%|█████████▊| 49/50 [04:20<00:05,  5.05s/it]

26 / 49


100%|██████████| 50/50 [04:25<00:00,  5.31s/it]

Expected: A
Got: B
26 / 50


0.52

We receive an accuracy of 0.52 on the tiny dataset